### DISCUSSION TITLE
Dynamic routes with params using data defetching fail on subsequent navigations

### DISCUSSION THREAD

**Discussion by minht11** [posva/unplugin-vue-router#169] [Questions and Help] [answered]
Using following page structure `route/[slug]` and using data fetching, changing between slugs `RangeError: Maximum call stack size exceeded` is thrown.

The whole issue is bit hard to explain, so here is a [reproduction](https://stackblitz.com/edit/vitejs-vite-w2tpkd?file=src%2Fpages%2Ftest%2F%5Bslug%5D%2Findex.vue)

1. Wait for the initial page to load.
2. Press "Test 2" link.
3. Observe error being thrown.

https://github.com/posva/unplugin-vue-router/assets/8914032/a1ac7286-0615-4e17-8db7-8c48d2c7e0a2

On my local setup if I on initial load/refresh everything is fine, only subsequent navigations result in failure.

**Comment by posva** [comment]
You had a few issues:

- Missing `setupDataFetchingGuard(router)`
- wrong options passed to `defineLoader()`

A few extra notes: you don't need to export the loader from App. In fact, App.vue is the only component that cannot use a loader as it mounts before the navigation starts (unless you await for the first navigation before calling `app.mount()`). Prefer using `{ name: ... }` when using router-link for type safety, passing the path directly is not type safe and it's easier to just pass the strirng in that case as well.

Here you have a working example: https://stackblitz.com/edit/vitejs-vite-pmdkkm?file=src%2FApp.vue,src%2Floader.ts,src%2Fpages%2Ftest%2F[slug]%2Findex.vue,src%2Fmain.ts (not sure how long it will stay up)

**Reply by minht11** [reply]
Sorry about missing setup, but the max range issue is still presant. But locally and inside the link you given, just press links back and forth and the range error should occur.



https://github.com/posva/unplugin-vue-router/assets/8914032/dd3acf32-252f-4c59-9c49-477d1fc3f510

**Reply by minht11** [reply]
@posva could you please check again. I still believe this is an actual issue and not the problem with the configuration, because range error is still present and easily reprodusable.

**Reply by posva** [reply, marked as answer]
https://github.com/posva/unplugin-vue-router/issues/173
